src/v1/function-configuration.ts

```diff
@@ -98,6 +98,11 @@ export const DEFAULT_FAILURE_POLICY: FailurePolicy = {
 
 export const MAX_NUMBER_USER_LABELS = 58;
 
+/**
+ * Invoker access control type for https functions.
+ */
+export type Invoker = 'public' | 'private' | string;
```

Comment:
A bit of a nit, but I prefer typing this as `'public' | 'private' | string | string[]`, and then making `RuntimeOptions.invoker: Invoker`
To me, this clarifies that something like`['private', 'public', 'my-service-acount@google.blah']` is not a valid option here.

Reply:
I (weakly) feel the opposite. One of the shorthands that was proposed was `'hosting'` and that one could make sense to be in the array.

Reply:
If we want to leave it as is for easier implementation of future work, I can do that. Alternatively, we can change it to Joe's suggestion and put a TODO comment when we add in the `hosting` feature. @joehan @inlined

Reply:
Sounds like its 2 votes to 1 here, and I don't think anyone feels super strongly, so probably leave it as is